c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -878,6 +879,7 @@ class DeviceCachingAllocator {
             TraceEntry::OOM,
             device_free,
             params.size(),
+            0, // not used for TraceEntry:OOM
```

Comment:
We have both the rounded and unrounded number here, so we should replace params.size() with the unrounded number and used rounded.size() for the rounded number. Previously we only reported the post-rounded number.

Reply:
Similar to `TraceEntry::ALLOC`, I'm now logging the unrounded size and also storing the `rounded_by` alongside it, to make it consistent and also align with the meaning of the newly added field.